modules/ensemble/lib/action/get_network_info_action.dart

```
@@ -0,0 +1,135 @@
+    super.inputs,
+    this.onSuccess,
+    this.onError,
+    this.onDenied,
+    this.onLocationDisabled});
+
+  factory GetNetworkInfoAction.from({Invokable? initiator,dynamic payload}) =>
+      GetNetworkInfoAction.fromYaml(initiator: initiator, payload: Utils.getYamlMap(payload));
+  factory GetNetworkInfoAction.fromYaml({Invokable? initiator, Map? payload}) {
+    return GetNetworkInfoAction(
+        initiator: initiator,
+        inputs: Utils.getMap(payload?['inputs']),
+        onSuccess: EnsembleAction.from(payload?['onSuccess'], initiator: initiator),
+        onError: EnsembleAction.from(payload?['onError'], initiator: initiator),
+        onDenied: EnsembleAction.from(payload?['onDenied'], initiator: initiator),
+        onLocationDisabled: EnsembleAction.from(payload?['onLocationDisabled'], initiator: initiator));
+  }
+
+  @override
+  Future<dynamic> execute(BuildContext context, ScopeManager scopeManager) {
+    networkInfo.getLocationStatus().then((locationStatus) {
+      if ((locationStatus == LocationPermissionStatus.denied.name
+          || locationStatus == LocationPermissionStatus.deniedForever.name)
+          && onDenied != null) {
+        return ScreenController().executeAction(context, onDenied!,
+            event: EnsembleEvent(initiator,
+                data: {'status': locationStatus}));
+      } else if (locationStatus == LocationStatus.disabled.name
+          && onLocationDisabled != null) {
+        return ScreenController().executeAction(context, onLocationDisabled!,
+            event: EnsembleEvent(initiator,
+                data: {'status': locationStatus}));
+      } else
+      if (locationStatus == LocationPermissionStatus.unableToDetermine.name
+          && onError != null) {
+        return ScreenController().executeAction(context, onError!,
+            event: EnsembleEvent(initiator,
+                data: {
+                  'status': locationStatus
+                }));
+      } else {
+        try {
```

You want to throw error if onSuccess is not specified? If you want to force it required, throw a LanguageError if not specified instead.

---

will do, thanks